[staging] json_c: 0.16 -> 0.17

panicgh opened the pull request:
## Description of changes

https://github.com/json-c/json-c/blob/json-c-0.17-20230812/ChangeLog

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

panicgh committed:
json-c: 0.16 -> 0.17

https://github.com/json-c/json-c/blob/json-c-0.17-20230812/ChangeLog

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3094

onemoresuza commented:
I'm getting the following error when running `nixpkgs-review pr 268526`:

>

SuperSandro2000 reviewed (commented):
we could add https://github.com/json-c/json-c/blob/HEAD/ChangeLog for meta.changelog

panicgh commented:
I've added meta.changelog and rebased the commit. Originally, I rebased it to a wrong branch, but fortunately, I marked the PR as draft just before and no mass spam-notification went out.

pbsds commented:
@ofborg eval

pbsds commented:
Sampled some reverse dependencies, and they all build


* [x] `nom build github:NixOS/nixpkgs/refs/pull/268526/head#json_c`
* [x] `nom build github:NixOS/nixpkgs/refs/pull/268526/head#mya`
* [x] `nom build github:NixOS/nixpkgs/refs/pull/268526/head#ttyd`
* [x] `nom build github:NixOS/nixpkgs/refs/pull/268526/head#clamav`
* [x] `nom build github:NixOS/nixpkgs/refs/pull/268526/head#drm_info`
* [x] `nom build github:NixOS/nixpkgs/refs/pull/268526/head#netifd`

LGTM
